motionplan/constraint.go

```diff
@@ -75,20 +88,20 @@ func resolveStatesToPositions(state *ik.State) error {
 }
 
 // SegmentFSConstraint tests whether a transition from a starting robot configuration to an ending robot configuration is valid.
-// If the returned bool is true, the constraint is satisfied and the segment is valid.
-type SegmentFSConstraint func(*ik.SegmentFS) bool
+// If the returned error is nil, the constraint is satisfied and the segment is valid.
+type SegmentFSConstraint func(*ik.SegmentFS) error
 
 // SegmentConstraint tests whether a transition from a starting robot configuration to an ending robot configuration is valid.
-// If the returned bool is true, the constraint is satisfied and the segment is valid.
-type SegmentConstraint func(*ik.Segment) bool
+// If the returned error is nil, the constraint is satisfied and the segment is valid.
+type SegmentConstraint func(*ik.Segment) error
```

Comment:
Weren't we going to remove these? Since we're making a breaking change anyway it's a great time

Reply:
That change takes this from a couple hours worth of change to a lot more than that.  Its going to take me some time to ramp on the TP space stuff to not mess that change up

Reply:
How about this.  I will timebox this change to no more than a few hours and if I cant do it then we will merge this?

Reply:
:+1:

Reply:
Ok I tried for a few hours this morning and didn't even get particularly close.  This is a larger request than I have time for

Reply:
Ok, reasonable.